axelor-account/src/main/java/com/axelor/apps/account/service/payment/invoice/payment/InvoicePaymentValidateServiceImpl.java

```diff
@@ -121,6 +126,16 @@ public void validate(InvoicePayment invoicePayment, boolean force)
     invoicePaymentRepository.save(invoicePayment);
   }
 
+  protected void checkConditionBeforeValidate(InvoicePayment invoicePayment)
+      throws AxelorException {
+    if (invoicePayment.getInvoice().getAmountRemaining().compareTo(BigDecimal.ZERO) <= 0) {
```

Comment:
Please check if it is possible to have invoice payment without invoice, because if not it will throw a NPE.